sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c

```diff
@@ -60,7 +60,43 @@ gnrc_tcp_tcb_t *_list_tcb_head;
 mutex_t _list_tcb_lock;
 
 /**
- * @brief   Establishes a new TCP connection.
+ * @brief Helper struct, holding all argument data for_cb_mbox_put_msg.
+ */
+typedef struct _cb_arg {
+    uint32_t msg_type;   /**< Message Type to Put into mbox behind mbox_ptr */
+    mbox_t *mbox_ptr;    /**< Pointer to mbox */
+} cb_arg_t;
+
+/**
+ * @brief Callback for xtimer, puts a message in a mbox.
+ *
+ * @param[in] arg   Ptr to cb_arg_t. Must not be NULL or anything else.
+ */
+static void _cb_mbox_put_msg(void *arg)
+{
+    msg_t msg;
+    msg.type = ((cb_arg_t *) arg)->msg_type;
+    mbox_try_put(((cb_arg_t *) arg)->mbox_ptr, &msg);
```

Comment:
I didn't dig any deeper into this, just some quick thoughts - likely I'm wrong, but that's why I'm asking you 😄